src/pyedb/dotnet/edb.py

```diff
@@ -4122,20 +4123,25 @@ def auto_parametrize_design(
             Enable padstack definition filter. Default value is ``None``, all padsatcks are parametrized.
         trace_net_filter : str, List(str), optional
             Enable nets filter for trace width parametrization. Default value is ``None``, all layers are parametrized.
+        use_single_variable_for_padstack_definitions : bool, optional
+            Whether to use a single design variable for each padstack definition or a variable per pad layer.
+            Default value is ``True``.
 
         Returns
         -------
         List(str)
             List of all parameters name created.
         """
+        if isinstance(trace_net_filter, str):
+            trace_net_filter = [trace_net_filter]
         parameters = []
         if layers:
             if not layer_filter:
-                _layers = self.stackup.stackup_layers
+                _layers = self.stackup.layers
```

Comment:
Why are we taking also non stackup layers ? I am nit sure we can apply parameters on them

Reply:
stackup_layers is deprecated
it returns layers